app/src/test/java/io/openshift/booster/SsoIT.java

```diff
@@ -0,0 +1,225 @@
+
+/**
+ * @author Heiko Braun
+ */
+public class SsoIT {
+    private static List<String> applicationUrls;
+
+    @BeforeClass
+    public static void setup() throws Exception {
+        OpenShiftClient oc = new DefaultOpenShiftClient();
+        List<Route> routes = oc.routes().inNamespace(oc.getNamespace()).list().getItems();
+
+        String ssoAuthUrl = routes.stream()
+                .filter(r -> "secure-sso".equals(r.getMetadata().getName()))
+                .findFirst()
+                .map(r -> "https://" + r.getSpec().getHost() + "/auth")
+                .orElseThrow(() -> new IllegalStateException("Couldn't find secure-sso route"));
+
+        System.setProperty("sso.auth.server.url", ssoAuthUrl);
+
+        authzClient = createAuthzCLient(ssoAuthUrl);
+
+        applicationUrls = routes.stream()
+                .filter(r -> r.getMetadata().getName().contains("secured"))
+                .map(r -> "http://" + r.getSpec().getHost())
+                .collect(toList());
+
+    }
+
+    private static Greeting getGreeting(String url, String token, String from) {
+        Client client = ClientBuilder.newClient();
+        try {
+            WebTarget target = client.target(url);
+            System.out.println("URL : " + url);
+            target.register((ClientRequestFilter) requestContext -> {
+                requestContext.getHeaders().add("Authorization", "Bearer " + token);
+            });
+            IGreeting greetingClient = ((ResteasyWebTarget) target).proxy(IGreeting.class);
+            return greetingClient.greeting(from);
+        } finally {
+            client.close();
+        }
+    }
+
+    @Test
+    public void defaultUser_defaultFrom() {
+        AccessTokenResponse accessTokenResponse = authzClient.obtainAccessToken("alice", "password");
+        for (String url : applicationUrls) {
+            Greeting greeting = getGreeting(url, accessTokenResponse.getToken(), null);
+
+            assertThat(greeting).isNotNull();
+            assertThat(greeting.getContent()).contains("Hello, World!");
+        }
+    }
+
+    @Test
+    public void defaultUser_customFrom() {
+        AccessTokenResponse accessTokenResponse = authzClient.obtainAccessToken("alice", "password");
+        for (String url : applicationUrls) {
+            Greeting greeting = getGreeting(url, accessTokenResponse.getToken(), "Scott");
+
+            assertThat(greeting).isNotNull();
+            assertThat(greeting.getContent()).contains("Hello, Scott!");
+        }
+    }
+
+  /*  @Test
+    public void adminUser() {
+        AccessTokenResponse accessTokenResponse = authzClient.obtainAccessToken("admin", "admin");
+        for (String url : applicationUrls) {
+            try {
+                getGreeting(url, accessTokenResponse.getToken(), null);
+                fail("ForbiddenException expected");
+            } catch (ForbiddenException e) {
+                // expected
+            }
+        }
+    }*/
```

Comment:
I thought this test passes now?

Reply:
I'll have to double check. But to me the assertion is unclear. Why would an admin user not be allowed to request a token?

Reply:
It's no longer `@Ignore`d in the SSO submodule.

And indeed the `admin` user is allowed to request a token, but doesn't have the correct role. This test checks the "authenticated, but not authorized" flow.

Reply:
I understand. Thanks for the clarification.